CMakeLists.txt

```diff
@@ -57,7 +57,7 @@ project (mongo-c-driver C)
 # Optionally enable C++ to do some C++-specific tests
 enable_language (CXX OPTIONAL)
 
-if (NOT CMAKE_BUILD_TYPE)
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
```

Comment:
Specifying `CMAKE_BUILD_TYPE` while using a multi-config generator doesn't change the behavior of CMake, but is Almost Always Wrong™, not in that it changes behavior of CMake (it doesn't), but because other code starts to assume that it is meaningful. Writing code that works on both single-config and multi-config is somewhat tricky.

Reply:
Thank you for the explanation.